File: test/rust/test-failed-assert.rs
fn fact(n: f32) -> f32 {
  let mut x = 1.0;
  for i in 0..n as i32 {
      x *= i as f32;
  }
  assert!(x != 0.0);
  x
}

#[chatdbg::main]
fn main() {
  println!("{}", fact(100.0));
}

File: samples/cpp/test-failed-assert.cpp
#include <assert.h>
#include <iostream>

float fact(float n) {
  auto x = 1.0;
  for (auto i = 0.0; i < n; i++) {
    x *= i;
  }
  assert(x != 0.0);
  return x;
}


int main()
{
  std::cout << fact(100) << std::endl;
  return 0;
}
